zellij-server/src/tab/mod.rs

```diff
@@ -1552,17 +1554,27 @@ impl Tab {
                         self.close_pane(PaneId::Terminal(active_terminal_id), false, None);
                         should_update_ui = true;
                     },
+                    Some(_) => {},
                     None => {},
                 }
             },
-            PaneId::Plugin(pid) => {
-                let mut plugin_updates = vec![];
-                for key in parse_keys(&input_bytes) {
-                    plugin_updates.push((Some(pid), client_id, Event::Key(key)));
-                }
-                self.senders
-                    .send_to_plugin(PluginInstruction::Update(plugin_updates))
-                    .with_context(err_context)?;
+            PaneId::Plugin(pid) => match active_terminal.adjust_input_to_terminal(input_bytes) {
+                Some(AdjustedInput::WriteBytesToTerminal(adjusted_input)) => {
+                    let mut plugin_updates = vec![];
+                    for key in parse_keys(&adjusted_input) {
+                        plugin_updates.push((Some(pid), client_id, Event::Key(key)));
+                    }
+                    self.senders
+                        .send_to_plugin(PluginInstruction::Update(plugin_updates))
+                        .with_context(err_context)?;
+                },
+                Some(AdjustedInput::Confirmed(_)) => {
+                    self.request_plugin_permissions(pid, None);
+                    // TODO: Maybe call render() in plugin instance?
```

Comment:
The UI is not updated immediately, and an update event must be generated from the plugin to call a new render and change the screen.

I think I have to force render to be called once. Is that correct?